src/main/java/io/cdap/e2e/pages/actions/WranglerActions.java

```diff
@@ -0,0 +1,260 @@
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.support.ui.Select;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * CDF wrangler related actions.
+ */
+public class WranglerActions {
+  private static final Logger logger = LoggerFactory.getLogger(WranglerActions.class);
+
+  static {
+    SeleniumHelper.getPropertiesLocators(WranglerLocators.class);
+  }
+
+  public static void selectDirective(String columnName, String directive) {
+    ElementHelper.clickOnElement(WranglerLocators.locateTransformationButton(columnName));
+    String pluginPropertyDirective = PluginPropertyUtils.getPluginPropertyElementTestId(directive);
+    ElementHelper.clickOnElement(WranglerLocators.locateDirectivesTitle(pluginPropertyDirective));
+  }
+
+  public static void selectDirectiveAndOption(String columnName, String directive, String option)
+    throws InterruptedException {
+    ElementHelper.clickOnElement(WranglerLocators.locateTransformationButton(columnName));
+    String pluginPropertyDirective = PluginPropertyUtils.getPluginPropertyElementTestId(directive);
+    switch (directive) {
+
+      case "CustomTransform":
+        ElementHelper.clickOnElement(WranglerLocators.locateDirectivesTitle(pluginPropertyDirective));
+        ElementHelper.sendKeysToTextarea(WranglerLocators.filterTextArea, option);
+        ElementHelper.clickOnElement(WranglerLocators.applyButton);
+        break;
+
+      case "Filter":
+      case "Sendtoerror":
+        ElementHelper.clickOnElement(WranglerLocators.locateDirectivesTitle(pluginPropertyDirective));
+        Select selectDropdown = new Select(WranglerLocators.filterSelect);
+        ElementHelper.scrollToElementUsingJsExecutor(WranglerLocators.selectDirectiveOption(option));
+        selectDropdown.selectByVisibleText(option);
+        ElementHelper.clickOnElement(WranglerLocators.applyButton);
+        break;
+
+      case "FillNullOrEmptyCells":
+      case "CopyColumn":
+        ElementHelper.clickOnElement(WranglerLocators.locateDirectivesTitle(pluginPropertyDirective));
+        ElementHelper.replaceElementValue(WranglerLocators.enterText, option);
+        ElementHelper.clickOnElement(WranglerLocators.applyButton);
+        break;
+
+      case "Hash":
+        ElementHelper.hoverOverElement(WranglerLocators.scrollButton);
+        Thread.sleep(3000);
+        ElementHelper.clickOnElement(WranglerLocators.locateDirectivesTitle(pluginPropertyDirective));
+        Select select = new Select(WranglerLocators.filterSelect);
+        ElementHelper.scrollToElementUsingJsExecutor(WranglerLocators.selectDirectiveOption(option));
+        select.selectByVisibleText(option);
+        ElementHelper.clickOnElement(WranglerLocators.applyButton);
+        break;
+
+      case "Decode":
+      case "Encode":
+        ElementHelper.hoverOverElement(WranglerLocators.scrollButton);
+        Thread.sleep(3000);
```

Comment:
Remove static sleep statements and replace it with dynamic ones. We should know what we are waiting for and for how long.

Reply:
done